### ISSUE TITLE
Undoing building type quest is not really working 

### ISSUE THREAD

**Issue by HolgerJeromin** [westnordost/StreetComplete#1131]
Latest (a few minutes) beta.

Doing an building type quest (answered as house) and a building level quest as this pops up.

Oh, this is in fact a garage.
Undo both => i see level quest and not type quest

**Comment by westnordost** [comment]
Right, quests, once created, are not removed again automatically when the underlying data changes. @matkoniecz , this is #746, only that combined with the undo-feature, in this configuration, it plays out as/like a bug.
